GroupManager/Utils/MSALPerUserMemoryTokenCache.cs

```diff
@@ -26,183 +26,182 @@
-        {
-            // Since we are using a MemoryCache ,whose methods are threads safe, we need not to do anything in this handler.
-        }
-
-        /// <summary>
-        /// To keep the cache, ClaimsPrincipal and Sql in sync, we ensure that the user's object Id we obtained by MSAL after
-        /// successful sign-in is set as the key for the cache.
-        /// </summary>
-        /// <param name="args">Contains parameters used by the MSAL call accessing the cache.</param>
-        private void SetSignedInUserFromNotificationArgs(TokenCacheNotificationArgs args)
-        {
-            if (this.SignedInUser == null && args.Account != null)
-            {
-                this.SignedInUser = args.Account.ToClaimsPrincipal();
-            }
-        }
-    }
+	/// <summary>
+	/// An implementation of token cache for both Confidential and Public clients backed by MemoryCache.
+	/// MemoryCache is useful in Api scenarios where there is no HttpContext to cache data.
+	/// </summary>
+	/// <seealso cref="https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/token-cache-serialization"/>
+	public class MSALPerUserMemoryTokenCache
+	{
+		/// <summary>
+		/// The backing MemoryCache instance
+		/// </summary>
+		internal readonly MemoryCache memoryCache = MemoryCache.Default;
+
+		/// <summary>
+		/// The duration till the tokens are kept in memory cache. In production, a higher value, upto 90 days is recommended.
+		/// </summary>
+		private readonly DateTimeOffset cacheDuration = DateTimeOffset.Now.AddHours(48);
+
+		/// <summary>
+		/// The internal handle to the client's instance of the Cache
+		/// </summary>
+		private ITokenCache UserTokenCache;
+
+		/// <summary>
+		/// Once the user signes in, this will not be null and can be ontained via a call to Thread.CurrentPrincipal
+		/// </summary>
+		internal ClaimsPrincipal SignedInUser;
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="MSALPerUserMemoryTokenCache"/> class.
+		/// </summary>
+		/// <param name="tokenCache">The client's instance of the token cache.</param>
+		public MSALPerUserMemoryTokenCache(ITokenCache tokenCache)
+		{
+			this.Initialize(tokenCache, ClaimsPrincipal.Current);
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="MSALPerUserMemoryTokenCache"/> class.
+		/// </summary>
+		/// <param name="tokenCache">The client's instance of the token cache.</param>
+		/// <param name="user">The signed-in user for whom the cache needs to be established.</param>
+		public MSALPerUserMemoryTokenCache(ITokenCache tokenCache, ClaimsPrincipal user)
+		{
+			this.Initialize(tokenCache, user);
+		}
+
+		/// <summary>
+		/// Explores the Claims of a signed-in user (if available) to populate the unique Id of this cache's instance.
+		/// </summary>
+		/// <returns>The signed in user's object Id , if available in the ClaimsPrincipal.Current instance</returns>
+		private string GetSignedInUsersUniqueId()
```

Comment:
>private string GetSignedInUsersUniqueId() [](start = 2, length = 41)

This is a dead method and removed in the other sample

Reply:
Method removed